### DISCUSSION TITLE
`HttpResponseMessage.RequestMessage` - when is this null?

### DISCUSSION THREAD

**Discussion by RenderMichael** [dotnet/runtime#104113] [Q&A] [answered]
Hi,

I am using the `HttpClient` to send messages, and I want to use the `HttpResponseMessage.RequestMessage` property.

However, the value is nullable. It's not clear to me when that is the case (other than a glib "it's null if you set it to null")

Can the `HttpClient.SendAsync` methods ever return a response message with a null request message?

**Comment by huoyaoyuan** [comment, marked as answer]
> Can the `HttpClient.SendAsync` methods ever return a response message with a null request message?

The default handlers will always set the property, but `HttpClient` support custom handlers which may not properly construct the response message.

**Reply by RenderMichael** [reply]
Ohhh I see, so letting ResponseMessage allow null is a defensive mechanism. Although having a null response message seems like a mistake to me (a mistake which customs handler implementers should be warned about) maybe I’m not seeing some scenario or maybe it would be a breaking change the team is unwilling to make.

Would it be OK for the docs to point this out? Anyone who wants to read the property is going to have to suppress a null or throw an errant exception, so it feels like a mainline scenario.

**Reply by huoyaoyuan** [reply]
The API shape of `HttpResponseMessage` was introduced far before nullable reference types. Constructing the object doesn't forces to pass a request message. You can always create a brand new instance yourself and pass it to some helper that processes the `RequestMessage` property.
